[tty] panic on termination

Hello there, on rare occasions, packer crashed for me:  upon termination + resizing the term windows at the same time: `panic: send on closed channel`.

I opened a pr here https://github.com/mattn/go-tty/pull/15/ to fix it.

It happening on windows too !
```
panic: send on closed channel

goroutine 1779 [running]:
github.com/hashicorp/packer/vendor/github.com/mattn/go-tty.(*TTY).readRune(0xc000305240, 0xc000000000, 0x0, 0x0)
	/Users/azr/go/src/github.com/hashicorp/packer/vendor/github.com/mattn/go-tty/tty_windows.go:209 +0xd33
github.com/hashicorp/packer/vendor/github.com/mattn/go-tty.(*TTY).readString(0xc000305240, 0x1, 0xc000326058, 0x30ce96d, 0x1, 0x0)
	/Users/azr/go/src/github.com/hashicorp/packer/vendor/github.com/mattn/go-tty/tty.go:60 +0x70
github.com/hashicorp/packer/vendor/github.com/mattn/go-tty.(*TTY).ReadString(0xc000305240, 0x0, 0x0, 0x0, 0x0)
	/Users/azr/go/src/github.com/hashicorp/packer/vendor/github.com/mattn/go-tty/tty.go:91 +0x95
github.com/hashicorp/packer/packer.(*BasicUi).Ask.func1(0xc0004262a0, 0xc000742600)
	/Users/azr/go/src/github.com/hashicorp/packer/packer/ui.go:212 +0x47
created by github.com/hashicorp/packer/packer.(*BasicUi).Ask
	/Users/azr/go/src/github.com/hashicorp/packer/packer/ui.go:211 +0x23f
```

--- a/go.mod
+++ b/go.mod
@@ -68,7 +68,7 @@ require (
 	github.com/masterzen/azure-sdk-for-go v0.0.0-20161014135628-ee4f0065d00c // indirect
 	github.com/masterzen/simplexml v0.0.0-20190410153822-31eea3082786 // indirect
 	github.com/masterzen/winrm v0.0.0-20180224160350-7e40f93ae939
-	github.com/mattn/go-tty v0.0.0-20190407112021-83fae09cc007
+	github.com/mattn/go-tty v0.0.0-20190424173100-523744f04859
 	github.com/miekg/dns v1.1.1 // indirect
 	github.com/mitchellh/cli v1.0.0
 	github.com/mitchellh/go-fs v0.0.0-20180402234041-7b48fa161ea7

--- a/main.go
+++ b/main.go
@@ -186,21 +186,19 @@ func wrappedMain() int {
 			return 1
 		}
 	} else {
-		var TTY packer.TTY
-		if !inPlugin {
-			var err error
-			TTY, err = tty.Open()
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "No tty available: %s\n", err)
-			} else {
-				defer TTY.Close()
-			}
-		}
-		ui = &packer.BasicUi{
+		basicUi := &packer.BasicUi{
 			Reader:      os.Stdin,
 			Writer:      os.Stdout,
 			ErrorWriter: os.Stdout,
-			TTY:         TTY,
+		}
+		ui = basicUi
+		if !inPlugin {
+			if TTY, err := tty.Open(); err != nil {
+				fmt.Fprintf(os.Stderr, "No tty available: %s\n", err)
+			} else {
+				basicUi.TTY = TTY
+				defer TTY.Close()
+			}
 		}
 	}
 	// Create the CLI meta

